astro/src/components/GitHubEdit.astro

```diff
@@ -0,0 +1,37 @@
+---
+import type { AstroGlobal } from "astro";
+
+/** Gets the URL to edit the page on GitHub */
+
+export function getGithubEditUrl(Astro: Readonly<AstroGlobal>) {
+  const slug = Astro.params?.slug;
+  const splitSlug = slug?.split("/");
+  const appendFile = splitSlug?.length
+    ? splitSlug?.length > 1
+      ? `/${slug}.md`
+      : `/${splitSlug.at(0)}/index.mdx`
+    : "";
+  const githubEditUrl = `https://github.com/FusionAuth/fusionauth-site/blob/master/astro/src/content/articles${appendFile}`;
```

Comment:
Does this work for quickstarts or devtools? I would guess not, since they are not under articles.

Reply:
those are all so different...lol

Updated to include all three types as they differ based on path.

Reply:
Hahah, for sure!

Can you please add a fallback because we'll be moving more content over and may forget to update this. I think a decent fallback would be to just link to the top level fusionauth-site repo.